### ISSUE TITLE
Script fails in worktrees

### ISSUE THREAD

**Issue by vhdlnerd** [capslocky/git-rebase-via-merge#2]
The script does not detect there were merge conflicts when used in a worktree.

The method to detect merge conflicts in `merge_conflicts_present()` does not work within worktrees.

**Comment by capslocky** [comment]
To be honest, I have never used git worktrees. If you could improve the script to work for your case I would be happy to merge your changes.

**Comment by vhdlnerd** [comment]
Alas, I'm not very good at Bash scripting -- I'm not sure if I could modify the script correctly.

However, I can give more details to help:
On a worktree, `.git` is not a directory but a normal text file (that's why `merge_conflicts_present()` fails). The `.git` file contains one line: the path to the main repo the worktree is based on.

So, if `.git` is not a directory (i.e., you are in a worktree) then `file_merge` in `merge_conflicts_present()` should be:
```bash
file_merge=contents_of_file($(git rev-parse --show-toplevel)/.git) + "/MERGE_HEAD"
```
I don't know how to do the `contents_of_file($(git rev-parse --show-toplevel)/.git)` part in Bash scripting.

**Comment by capslocky** [comment]
I see, you can try to modify that method like this
```bash
merge_conflicts_present() {
  dot_git="$(git rev-parse --show-toplevel)/.git"
  echo dot_git=$dot_git

  if [ -d "$dot_git" ]; then
    # .git is directory
    file_merge="$dot_git/MERGE_HEAD"
  else
    # .git is file (worktree)
    file_merge="`cat "$dot_git"`/MERGE_HEAD"
  fi

  echo file_merge=$file_merge
  [ -e "$file_merge" ]
}
```
I didn't test it with a real worktree though, so you might need to fix it. Please post back the working code or you can even raise a pull-request.

**Comment by vhdlnerd** [comment]
I had to change the code a little.  The path in `.git` has a prefix of `gitdir: ` that needed to be stripped (I missed that before).  The following works great:

```bash
merge_conflicts_present() {
  dot_git="$(git rev-parse --show-toplevel)/.git"
  echo dot_git=$dot_git

  if [ -d "$dot_git" ]; then
    # .git is directory
    file_merge="$dot_git/MERGE_HEAD"
  else
    # .git is file (worktree)
    file_merge="`sed -n -e 's/^gitdir: //p' $dot_git`/MERGE_HEAD"
  fi

  echo file_merge=$file_merge
  [ -e "$file_merge" ]
}
```

BTW: I'm using/testing your script where I work. I cannot push to github from my place of work. So, I cannot create a PR. :disappointed:

**Comment by vhdlnerd** [comment]
Also, there is one more issue. The script fails if there is more than one rebase conflict.  I used the script on a rebase that had about 100 commits (!).  During the rebase phase three of the commits caused a conflict. (I don't think this would happen often but it had to do with files being renamed and `git` not auto detecting the file name changes.)

In `fix_rebase_conflicts()` I changed/added:
```bash
    if [ "$input" = "c" ]; then
      if [ -z "$(get_unstaged_files)" ]; then
        git rebase --continue  || true
        if ! rebase_conflicts_present; then
          break
        fi
```
Added the `|| true` and the `rebase_conflicts_present` test. This allowed `git rebase --continue` to be executed over and over until no conflicts.
